bark/commons/tests/frenet_test.cc

```diff
@@ -125,4 +126,83 @@ TEST(frenet_state_two_way, straight_line_top_right) {
 
   // state on right side of path with orientation on path
   // test_state_two_way(-1, 5, B_PI_2, 5, line);
-}
\ No newline at end of file
+}
+
+TEST(transform_lat_acc_street_to_vehicle, straight_line_aligned) {
+  Line line;
+  line.AddPoint(Point2d(0, 0));
+  line.AddPoint(Point2d(10, 0));
+
+  double acc_lat_street = 0.1;
+  double acc_lon = 0.0;
+  double delta_time = 1;
+
+  State last_state(static_cast<int>(st::MIN_STATE_SIZE));
+  last_state << 0, 0, 0, 0, 1;
+  State current_state(static_cast<int>(st::MIN_STATE_SIZE));
+  current_state << delta_time,
+      delta_time * last_state(StateDefinition::VEL_POSITION), 0, 0, 1;
+
+  FrenetState last_frenet_state(last_state, line);
+  FrenetState current_frenet_state(current_state, line);
+
+  double acc_lat = LatAccStreetToVehicleCs(acc_lat_street, acc_lon, delta_time,
+                                           current_state, current_frenet_state,
+                                           last_frenet_state);
+
+  EXPECT_EQ(acc_lat, acc_lat_street);
+}
+
+TEST(transform_lat_acc_street_to_vehicle, left_curved_line_aligned) {
+  Line line;
+  line.AddPoint(Point2d(0, 0));
+  line.AddPoint(Point2d(1, 0));
+  line.AddPoint(Point2d(2, 1));
+  line.AddPoint(Point2d(2, 10));
+
+  double acc_lat_street = 0.1;
+  double acc_lon = 0.0;
+  double delta_time = 1;
+
+  State last_state(static_cast<int>(st::MIN_STATE_SIZE));
+  last_state << 0, 0, 0, 0, 1;
+  State current_state(static_cast<int>(st::MIN_STATE_SIZE));
+  current_state << delta_time,
+      delta_time * last_state(StateDefinition::VEL_POSITION), 0, 0, 1;
+
+  FrenetState last_frenet_state(last_state, line);
+  FrenetState current_frenet_state(current_state, line);
+
+  double acc_lat = LatAccStreetToVehicleCs(acc_lat_street, acc_lon, delta_time,
+                                           current_state, current_frenet_state,
+                                           last_frenet_state);
+
+  EXPECT_GT(acc_lat, acc_lat_street);
+}
+
+TEST(transform_lat_acc_street_to_vehicle, right_curved_line_aligned) {
+  Line line;
+  line.AddPoint(Point2d(0, 0));
+  line.AddPoint(Point2d(1, 0));
+  line.AddPoint(Point2d(2, -1));
+  line.AddPoint(Point2d(2, -10));
+
+  double acc_lat_street = 0.1;
+  double acc_lon = 0.0;
+  double delta_time = 1;
+
+  State last_state(static_cast<int>(st::MIN_STATE_SIZE));
+  last_state << 0, 0, 0, 0, 1;
+  State current_state(static_cast<int>(st::MIN_STATE_SIZE));
+  current_state << delta_time,
+      delta_time * last_state(StateDefinition::VEL_POSITION), 0, 0, 1;
+
+  FrenetState last_frenet_state(last_state, line);
+  FrenetState current_frenet_state(current_state, line);
+
+  double acc_lat = LatAccStreetToVehicleCs(acc_lat_street, acc_lon, delta_time,
+                                           current_state, current_frenet_state,
+                                           last_frenet_state);
+
+  EXPECT_LT(acc_lat, acc_lat_street);
```

Comment:
why is the acc_lat also 0.1 here? should it not be reversed to the left_curved test?